lib/runtime/src/ops/split.cc

```diff
@@ -52,357 +53,126 @@ SplitParams Split::get_params() const {
-  runtime->execute_index_space(ctx, launcher);
+  return {SPLIT_INIT_TASK_ID, binding};
 }
 
-PerDeviceOpState *Split::init_task(Task const *task,
-                                   std::vector<PhysicalRegion> const &regions,
-                                   Context ctx,
-                                   Runtime *runtime) {
-  return NULL;
+OpTaskInvocation forward(SplitAttrs const &attrs) {
+  OpTaskBinding binding;
+
+  binding.bind_arg(PROFILING, profiling_settings());
+  binding.bind(INPUT, input_parallel_tensor_shape(0));
+  binding.bind(OUTPUT, output_tensor(0));
+
+  return {SPLIT_FWD_TASK_ID, binding};
 }
 
-void Split::forward(FFModel const &ff) {
-  ArgumentMap argmap;
-  Context ctx = ff.config.lg_ctx;
-  Runtime *runtime = ff.config.lg_hlr;
-  IndexLauncher launcher(SPLIT_FWD_TASK_ID,
-                         parallel_is,
-                         TaskArgument(this, sizeof(Split)),
-                         argmap,
-                         Predicate::TRUE_PRED,
-                         false /*must*/,
-                         0 /*mapper_id*/,
-                         outputs[0]->machine_view.hash());
-  launcher.add_region_requirement(RegionRequirement(inputs[0]->part,
-                                                    0 /*projection id*/,
-                                                    READ_ONLY,
-                                                    EXCLUSIVE,
-                                                    inputs[0]->region));
-  launcher.add_field(0, FID_DATA);
-  for (int i = 0; i < numOutputs; i++) {
-    launcher.add_region_requirement(RegionRequirement(outputs[i]->part,
-                                                      0 /*projection id*/,
-                                                      WRITE_ONLY,
-                                                      EXCLUSIVE,
-                                                      outputs[i]->region));
-    launcher.add_field(i + 1, FID_DATA);
-  }
-  runtime->execute_index_space(ctx, launcher);
+OpTaskInvocation backward(SplitAttrs const &attrs) {
+  OpTaskBinding binding = infer_bwd_binding(forward(attrs).binding);
+
+  return {SPLIT_BWD_TASK_ID, binding};
 }
 
-void calc_block_size(coord_t &num_blks,
-                     coord_t &blk_size,
-                     Domain const &domain,
-                     int axis) {
-  num_blks = 1;
-  blk_size = 1;
-  for (int d = 0; d < domain.get_dim(); d++) {
-    if (d <= axis) {
-      blk_size *= (domain.hi()[d] - domain.lo()[d] + 1);
-    } else {
-      num_blks *= (domain.hi()[d] - domain.lo()[d] + 1);
-    }
-  }
+static optional<float> forward_task_impl(TaskArgumentAccessor const &acc) {
+  acc.get_argument<CastPerDeviceState>(PER_DEVICE_STATE);
+  ProfilingSettings profiling = acc.get_argument<ProfilingSettings>(PROFILING);
+
+  auto input = acc.get_tensor<Permissions::RO>(INPUT);
+  auto output = acc.get_tensor<Permissions::WO>(OUTPUT);
+
+  // Note: forward_kernel needs parameter Legion::coord_t const
```

Comment:
how to get `parameter Legion::coord_t const
  // *out_blk_sizes,Legion::coord_t in_blk_size, Legion::coord_t num_blks, int
  // numOutputs how to get these parameter?`

Reply:
```
void calc_block_size(coord_t &num_blks,
                     coord_t &blk_size,
                     Domain const &domain,
                     int axis) {
  num_blks = 1;
  blk_size = 1;
  for (int d = 0; d < domain.get_dim(); d++) {
    if (d <= axis) {
      blk_size *= (domain.hi()[d] - domain.lo()[d] + 1);
    } else {
      num_blks *= (domain.hi()[d] - domain.lo()[d] + 1);
    }
  }
}
```

This is the code in the original operator code. I think you can change this to take `ArrayShape` instead of `Domain`.